src/Language/PureScript/Renamer.hs

```diff
@@ -175,7 +177,7 @@ renameInLiteral _ l = return l
 -- Renames within case alternatives.
 --
 renameInCaseAlternative :: CaseAlternative Ann -> Rename (CaseAlternative Ann)
-renameInCaseAlternative (CaseAlternative bs v) =
+renameInCaseAlternative (CaseAlternative bs v) = newScope $
```

Comment:
No I thought so, but you were right - it was the case desugaring. There are legitimate cases where we cause overlaps, but I figure it's ok, since we know where they happen and can ensure it's safe.